pkg/credentials/credentials.go

```diff
@@ -158,6 +160,8 @@ func ProviderCredentials(cloudProvider kubeone.CloudProviderSpec, credentialsFil
 			{Name: OpenStackAuthURL},
 			{Name: OpenStackUserName, MachineControllerName: OpenStackUserNameMC},
 			{Name: OpenStackPassword},
+			{Name: OpenStackApplicationCredentialID, MachineControllerName: OpenStackUserNameMC},
```

Comment:
`MachineControllerName` is not needed here:

```suggestion
			{Name: OpenStackApplicationCredentialID},
```